### PR TITLE
API - Add Delete to Debtors Note

### PR DESCRIPTION
**Pull request by caismith-ch**
Add the ability to delete a debtors note and return the correct response statuses.

**Resolves**: SFA-990

### CODE REVIEW

**Comment by BSaunders95** [comment]
For the delete functionality, will we be removing the small full resource link?

**Comment by caismith-ch** [comment]
Yeah I think you are right there, Will look into doing that.

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java [hunk 1]
```diff
@@ -242,9 +238,7 @@ void updateAccountingPoliciesSuccess() throws DataException {
     @DisplayName("Update debtors - data exception thrown")
     void updateAccountingPoliciesDataException() throws DataException {
 
-        when(mockRequest.getAttribute(anyString())).thenReturn(mockSmallFull).thenReturn(mockTransaction);
-        when(mockSmallFull.getLinks()).thenReturn(mockSmallFullLinks);
-        when(mockSmallFullLinks.get(SmallFullLinkType.DEBTORS_NOTE.getLink())).thenReturn("");
+        setupTransactionAndLinks();
```
Comment (ch-mcabrera) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/DebtorsControllerTest.java]: I would rename this to `mockTransactionAndLinks()`.

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java [hunk 2]
```diff
@@ -134,6 +134,24 @@ public DebtorsService(DebtorsRepository repository,
         return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(entity));
     }
 
+    @Override
+    public ResponseObject<Debtors> deleteById(String id, HttpServletRequest request) throws DataException {
+        if (repository.existsById(id)) {
```
Comment (ch-mcabrera) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java]: I would move this statement to the `try and catch` as `repository.existsById(id)` throws a MongoException

```
try {
    if (repository.existsById(id)) {
        repository.deleteById(id);
        return new ResponseObject<>(ResponseStatus.UPDATED);
    } else {
        return new ResponseObject<>(ResponseStatus.NOT_FOUND);
    }

} catch (MongoException me) {
    ...
}
```

**Pull request merged**